### PR TITLE
Support compiling C modules that don't use wasi

### PR DESCRIPTION
**Pull request by torch2424**
This adds support for compiling C files that don't use WASI, to Wasm.

This is done by essentially matching the arguments used by [this article](https://depth-first.com/articles/2019/10/16/compiling-c-to-webassembly-and-running-it-without-emscripten/).

And here is the current example I am compiling: https://github.com/torch2424/wasmer-examples/tree/master/c/hello-world/example-wasienv-wasm/add-one

# Pending Items

* The current API is by running something like: `wasicc add-one.c --no-wasi -o add-one`. No wasi is a weird flag name, thus, let's discuss!
* Currently, it makes all functions available to the user, we should find how to select which functions to export from the linked.

# Additional Resources 

https://aransentin.github.io/cwasm/

# Example

Screen Shot 2019-11-06 at 10 58 01 AM
Screen Shot 2019-11-06 at 10 57 50 AM

### CODE REVIEW

File: wasienv/wasicc.py [hunk 1]
```diff
@@ -24,16 +24,29 @@ def run(args):
...
+    # Remove the no wasi from the args since clang doesn't support it
+    if "--no-wasi" in args: args.remove("--no-wasi")
+
+    if has_no_wasi:
+        args.append("--no-standard-libraries")
+        args.append("-Wl,--export-all")
+        args.append("-Wl,--no-entry")
+
+   
+    if not has_no_wasi:
```
Comment (syrusakbary) [inline wasienv/wasicc.py]: Can you do an `else` here?

Comment (torch2424) [reply wasienv/wasicc.py]: Yeah sure! 😀

**Comment by torch2424** [comment]
Per a discussion on the team slack, we decided it would be better to have both a `wasmcc` and `wasicc` command, as this would avoid confusing with "negative" flags that remove imports and things.

**Comment by torch2424** [comment]
@syrusakbary This is good to go! The test that failed was during install for a network error, so it just needs to be re-run 😄

**Comment by torch2424** [comment]
For future refence, llvm linker docs: https://lld.llvm.org/WebAssembly.html

**Pull request merged**